modules/convert/pull.go

```diff
@@ -99,33 +101,35 @@ func ToAPIPullRequest(pr *models.PullRequest) *api.PullRequest {
 		apiPullRequest.Base = apiBaseBranchInfo
 	}
 
-	headBranch, err = repo_module.GetBranch(pr.HeadRepo, pr.HeadBranch)
-	if err != nil {
-		if git.IsErrBranchNotExist(err) {
-			apiPullRequest.Head = nil
-		} else {
-			log.Error("GetBranch[%s]: %v", pr.HeadBranch, err)
-			return nil
-		}
-	} else {
-		apiHeadBranchInfo := &api.PRBranchInfo{
-			Name:       pr.HeadBranch,
-			Ref:        pr.HeadBranch,
-			RepoID:     pr.HeadRepoID,
-			Repository: pr.HeadRepo.APIFormat(models.AccessModeNone),
-		}
-		headCommit, err = headBranch.GetCommit()
+	if pr.HeadRepo != nil {
```

Comment:
In the case that the head repo is nil maybe we need to give some placeholder information (?) - similar to what the UI gives?

Reply:
you mean implement a `ghost repo` :D

Reply:
Maybe?

```go
apiHeadBranchInfo := &api.PRBranchInfo{
    Name: pr.HeadBranch,
    Ref: pr.HeadBranch,
    RepoID: -1,
} 
```

Reply:
In the UI we should really be reporting the repo as "(deleted)"